### ISSUE TITLE
form is not resetting its value

### ISSUE THREAD

**Issue by fazalrasel** [christianalfoni/formsy-react#396]
## my submit function
```
   submit = (values) => {
      return axios
         .post('shops', values)
         .then((res) => {
            alert.success('Shop Created.'); // on success, its showing
            this.refs.newShopForm.reset(); // but this is not setting forms value
         })
         .catch((err) => {
            this.refs.newShopForm.updateInputsWithError(err.response.data); // on error, its showing errors
         });
   };
```

**Comment by hschoenburg** [comment]
I'm having the same issue. Can't get the form to reset

**Comment by hschoenburg** [comment]
Cannot for the life of me get the from to reset its values.
Trying:

```
this.refs.form.reset();
```

Tried passing value={this.state.something} and then calling this.setState({value: ''}) on submit. That didn't work either.

**Comment by acomito** [comment]
same here

**Comment by klanger** [comment]
Me too, using formsy-material-ui. And I'm stuck when it comes to debugging it too.

**Comment by aun343** [comment]
Me too.
I'm not having the issue with react v0.14.x but having it used with react 15.x

**Comment by aesopwolf** [comment]
Are you guys using `getValue()` on your custom input element?

> [You should always use the getValue() method inside your formsy form element.](https://github.com/christianalfoni/formsy-react/blob/master/API.md#value)

**Comment by brianfegan** [comment]
In my experience it resets the formsy val, but not the values of the form elements.

**Comment by prasanna1211** [comment]
I agree with that. Resetting the form values is not same as resetting the form elements. Writing the formsyComponents in such a way that resetting the values using this.refs.form.reset() will reset the value of the element thereby forcing the form elements to change would do.

**Comment by aesopwolf** [comment]
@brianfegan and @prasanna1211 would you mind providing sample code? In my experience `reset` works fine but only when you use `getValue()` in your component.

**Comment by brianfegan** [comment]
Where would the getValue() go in the form element component? Currently, I'm just using setValue inside onChange event handlers.

To actually reset the HOC form element value, I pass a callback into HOC form element from the formsy form component that gets called when the HOC form element mounts. That formsy form callback has an  arg that's a ref to a custom HOC form element reset method callback to setValue within the form element component.

It works but its not the cleanest implementation for sure.

**Comment by aesopwolf** [comment]
@brianfegan as the `value` attribute itself https://github.com/christianalfoni/formsy-react/blob/master/examples/components/Input.js#L35

Try updating your component to use `getValue` and let me know how it goes.

**Comment by brianfegan** [comment]
Yeah, I'm using that for individual text inputs. But the issue I described deals with multi checkbox values.

**Comment by brianfegan** [comment]
...though I suppose based on what you said, the custom reset method in the HOC could call getValue to repopulate the state in the HOC component (currently I pass the new value in as an arg and don't use getValue).

**Comment by aesopwolf** [comment]
@brianfegan Yeah you really should be using `getValue`. Would you mind setting up a demo repo of the problem you're facing?

**Comment by brianfegan** [comment]
I'm using a very slightly modified version of https://github.com/christianalfoni/formsy-react/blob/master/examples/components/MultiCheckboxSet.js. Just added some markup for styling. As you can see, there's no getValue in that example.

**Comment by C3PablO** [comment]
I had the same problem but just with textarea fields. I have a formsy-react wrapper for my input components so I can use them with o without formsy-react. Calling reset was cleaning the values as expected but passing `undefined` as a value to the children component didn't updated it so I end up passing `value={this.getValue() || ''}` and that solved the issue

**Comment by cristianocca** [comment]
So is there a way to manually set the value of an input of a FormsyText component and have the actual form input get updated?
I have tried setValue through a very ugly way ( something like `this.refs.form.inputs[this.refs.form.prevInputNames.indexOf("name")].setValue("TEST") `) and the value is updated when getting it back, for the form UI is never updated, there are no visual changes.

How can this be achieved easier?
